### PR TITLE
Allow workbenches to keep the image digest up-to-date

### PR DESCRIPTION
**Pull request by harshad16**
Allow workbenches to keep the image digest up-to-date

## Description

Related-to: https://github.com/opendatahub-io/odh-manifests/issues/829

Openshit allow periodic pull of image digest from it external source 
https://docs.openshift.com/container-platform/4.12/openshift_images/image-streams-manage.html#images-imagestream-import_image-streams-managing
https://docs.openshift.com/container-platform/3.11/dev_guide/managing_images.html#importing-tag-and-image-metadata

with the `scheduled` flag , if the user has out dated sha of the weekly tag
this would help out.

## How Has This Been Tested?

Setup the ODH cluster
Try out the new image tag and update the tag
scheduled will pull the latest tag.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by atheo89 — COMMENTED** [review]
Nice catch, I think `scheduled` is what we need here
/lgtm

**Comment by shalberd** [comment]
Yup: https://github.com/opendatahub-io/kubeflow/pull/72#issuecomment-1496946371

**Comment by shalberd** [comment]
suggestion:

For non-dev purposes, keep the scheduled: true in the weekly tags out of the imagestreams. That is, keep that out of any branch other than master, e.g. v1.6

On the other hand, on airgapped clusters, scheduled: true will not work anyways with regards to looking at the external registry for changes to the weekly data science image digests ...

File: notebook-images/base/jupyter-minimal-gpu-notebook-imagestream.yaml [hunk 1]
```diff
@@ -23,6 +23,8 @@ spec:
       kind: DockerImage
       name: quay.io/opendatahub/workbench-images:cuda-jupyter-minimal-ubi9-python-3.9-2023a-weekly
```
Comment (shalberd) [inline notebook-images/base/jupyter-minimal-gpu-notebook-imagestream.yaml:24]: very good idea to keep the from.name image reference to the source-repo tag-based on the master branch.

File: notebook-images/base/jupyter-minimal-gpu-notebook-imagestream.yaml [hunk 2]
```diff
@@ -23,6 +23,8 @@ spec:
...
     name: "2023.1-cuda-11.8"
+    importPolicy:
+      scheduled: true
     referencePolicy:
       type: Local
```
Comment (shalberd) [inline notebook-images/base/jupyter-minimal-gpu-notebook-imagestream.yaml:29]: change referencePolicy to `type: Source `to be independent of the internal openshift registry. I think we can safely assume quay.io will always be available :-)

File: notebook-images/base/jupyter-minimal-gpu-notebook-imagestream.yaml [hunk 3]
```diff
@@ -23,6 +23,8 @@ spec:
       kind: DockerImage
       name: quay.io/opendatahub/workbench-images:cuda-jupyter-minimal-ubi9-python-3.9-2023a-weekly
     name: "2023.1-cuda-11.8"
+    importPolicy:
+      scheduled: true
```
Comment (shalberd) [inline notebook-images/base/jupyter-minimal-gpu-notebook-imagestream.yaml:27]: yes, that setting for the weekly-changing tags assures that the imagestream status-section status.tags[x].items.dockerImageReference always refers to the latest digest value, as well as status.tags[x].items.image.

**Pull request closed without merging**